impl/command_line/options.cxx

```diff
@@ -128,6 +131,29 @@ namespace substrate::commandLine
 		return std::nullopt;
 	}
 
+	// This calculates how much padding is needed for these options to get a consistent padding
+	[[nodiscard]] size_t optionSet_t::displayPadding() const noexcept
+	{
+		return std::accumulate(begin(), end(), size_t{},
+			[](const size_t padding, const optionAlternation_t &alternation) noexcept
+				{ return std::max(padding, alternation.displayLength()); }
+		);
```

Comment:
If you have a working example, I'd be interested in comparing codegens against:

```suggestion
		return std::max_element(begin(), end(), 
			[](const optionAlternation_t &lhs, const optionAlternation_t &rhs) noexcept
				{ return std::max(lhs.displayLength(), rhs.displayLength()); }
		);
```

Reply:
Certainly, compiling [bmpflash](https://github.com/blackmagic-debug/bmpflash) against this PR will provide a suitable test binary to inspect as that uses this new subsystem to [drive its help generation](https://github.com/blackmagic-debug/bmpflash/blob/5df2c1bb51270f7b26ea9269a9670e15de53fae9/src/bmpflash.cxx#L28-L39)

Reply:
Doesn't appear to affect codegen, and it's more readable than the `max_element` approach in the face of the visitor pattern in `optionsHolder_t`.